lib/modules/versioning/types.ts

```diff
@@ -10,26 +10,88 @@ export interface NewValueConfig {
 }
 export interface VersioningApi {
   // validation
+
+  /**
+   * Check whether the `version` is compatible with the `current` version range
+   * constraint.
+   */
   isCompatible(version: string, current?: string): boolean;
+
+  /**
+   * Check whether the `version` constraint is pinned, i.e. it only allows a
+   * single specific version.
+   */
   isSingleVersion(version: string): boolean;
+
+  /**
+   * Check whether the `version` is considered to be "stable".
+   *
+   * Example: in SemVer the version must not have a pre-release marker.
+   */
   isStable(version: string): boolean;
+
+  /**
+   * Check whether the `input` is a valid version range constraint.
```

Comment:
if a version or range is valid for the current versioning scheme

Reply:
```suggestion
   * Check whether the `input` is a valid version or a valid version range constraint.
```